poi-ooxml/src/test/java/org/apache/poi/xssf/usermodel/TestXSSFWorkbook.java

```diff
@@ -1299,6 +1296,50 @@ void testIgnoreTrashParts() throws Exception {
         }
     }
 
+    @Test
+    void testCacheExternalWorkbook() throws Exception {
+        String nameA = "cache-external-workbook-a.xlsx";
+
+        try (
+                UnsynchronizedByteArrayOutputStream bosA = new UnsynchronizedByteArrayOutputStream();
+                UnsynchronizedByteArrayOutputStream bosB = new UnsynchronizedByteArrayOutputStream();
+                XSSFWorkbook workbookA = new XSSFWorkbook();
+                XSSFWorkbook workbookB = new XSSFWorkbook()
+        ) {
+            XSSFRow row1 = workbookA.createSheet().createRow(0);
+            double v1 = 10, v2 = 10, sum = v1 + v2;
+            row1.createCell(0).setCellValue(v1);
+            row1.createCell(1).setCellValue(v2);
+
+            XSSFRow row = workbookB.createSheet().createRow(0);
+            XSSFCell cell = row.createCell(0);
+
+            workbookB.linkExternalWorkbook(nameA, workbookA);
+            String formula = String.format(LocaleUtil.getUserLocale(), "SUM('[%s]Sheet0'!A1:B1)", nameA);
+            cell.setCellFormula(formula);
+            XSSFFormulaEvaluator evaluator = workbookB.getCreationHelper().createFormulaEvaluator();
+            evaluator.evaluateFormulaCell(cell);
+
+            assertEquals(sum, cell.getNumericCellValue());
+
+            workbookA.write(bosA);
+            workbookB.write(bosB);
+            FileOutputStream f1 = new FileOutputStream(nameA);
```

Comment:
please avoid writing files in tests, use UnsynchronizedByteArrayOutputStream instead